Content.Client/Chemistry/UI/ChemMasterWindow.xaml.cs

```diff
@@ -180,38 +181,63 @@ private void UpdatePanelInfo(ChemMasterBoundUserInterfaceState state)
+                    layerCount++;
+                    Color currentLayerColor = (layerCount % 2 == 1) ? layerColor1 : layerColor2;
+                    
+                    BufferInfo.Children.Add(new PanelContainer
                     {
-                        Orientation = LayoutOrientation.Horizontal,
+                        // Applies a color tint to the current reagent row
+                        PanelOverride = new StyleBoxFlat(currentLayerColor),
                         Children =
                         {
-                            new Label {Text = $"{name}: "},
-                            new Label
+                            new BoxContainer
                             {
-                                Text = $"{quantity}u",
-                                StyleClasses = {StyleNano.StyleClassLabelSecondaryColor}
-                            },
-
-                            // Padding
-                            new Control {HorizontalExpand = true},
-
-                            MakeReagentButton("1", ChemMasterReagentAmount.U1, reagent, true, StyleBase.ButtonOpenRight),
-                            MakeReagentButton("5", ChemMasterReagentAmount.U5, reagent, true, StyleBase.ButtonOpenBoth),
-                            MakeReagentButton("10", ChemMasterReagentAmount.U10, reagent, true, StyleBase.ButtonOpenBoth),
-                            MakeReagentButton("25", ChemMasterReagentAmount.U25, reagent, true, StyleBase.ButtonOpenBoth),
-                            MakeReagentButton("50", ChemMasterReagentAmount.U50, reagent, true, StyleBase.ButtonOpenBoth),
-                            MakeReagentButton("100", ChemMasterReagentAmount.U100, reagent, true, StyleBase.ButtonOpenBoth),
-                            MakeReagentButton(Loc.GetString("chem-master-window-buffer-all-amount"), ChemMasterReagentAmount.All, reagent, true, StyleBase.ButtonOpenLeft),
+                                Orientation = LayoutOrientation.Horizontal,
+                                Children =
+                                {
+                                    new Label { Text = $"{name}: " },
+                                    new Label
+                                    {
+                                        Text = $"{quantity}u",
+                                        StyleClasses = { StyleNano.StyleClassLabelSecondaryColor }
+                                    },
+
+                                    // Padding
+                                    new Control { HorizontalExpand = true },
+                                    // Provides a Unicode "medium vertical bar" and colors it based on the reagent.
+                                    new Label
+                                    {
+                                        Text = "\u2759",
+                                        Modulate = proto.SubstanceColor
+                                    },
```

Comment:
Should be a non-label Control like Chemical Dispenser

And huh, ChemMaster has proto stuff in the `.xaml.cs` already? Damn